examples/03-javascripts-objects-classes/mars_rover.js

```diff
@@ -13,7 +13,101 @@
 //
 
 class MarsRover {
+  constructor(posX, posY) {
+    this.x = posX;
+    this.y = posY;
+    this.direction = 'north';
+  }
+
+  // gets and sets the position
+  position() {
+    return {
+      x: this.x,
+      y: this.y
+    }
+  }
+
+  //gets the direction
+  get_direction() {
+    return this.direction;
+  }
+
+  move() {
+    switch(this.get_direction()) {
```

Comment:
@jlabausa I like this. Using the method instead of the attribute allows you to cascade changes into `move()` in case the `direction()` definition changes.